lib/queue.js

```diff
@@ -1,58 +1,216 @@
+
+  scheduled(options) {
+    const offset = (options && options.offset) || 0;
+    const count = (options && options.count) || 25;
+    return this.client.call('jobs', 'scheduled', this.name, offset, count);
+  }
+
+  depends(options) {
+    const offset = (options && options.offset) || 0;
+    const count = (options && options.count) || 25;
+    return this.client.call('jobs', 'depends', this.name, offset, count);
+  }
+
+  recurring(options) {
+    const offset = (options && options.offset) || 0;
+    const count = (options && options.count) || 25;
+    return this.client.call('jobs', 'recurring', this.name, offset, count);
+  }
+
+}
 
 class Queue {
-  constructor(name, client, workerName) {
+
+  constructor(name, client) {
     this.name = name;
     this.client = client;
-    this.workerName = workerName; // TODO -- might want to get from client, but given how JS works (more like greenlets than ruby 1 process), probably ok now
-    // "_hb" are in Python only -- what are these?
+    this.jobs = new Jobs(name, client);
   }
 
-  running(start, count, cb) {
-    this.client.call('jobs', 'running', this.name, start || 0, count || 25, cb);
+  static makeJobId() {
+    return uuid.v4().replace(/-/g, '');
   }
-  stalled(start, count, cb) {
-    this.client.call('jobs', 'stalled', this.name, start || 0, count || 25, cb);
+
+  getHeartbeat() {
+    const key = `${this.name}-heartbeat`;
+    return this.client.getConfig()
+      .then(config => (config[key] || config.heartbeat));
+  }
+
+  setHeartbeat(value) {
+    return this.client.setConfig(`${this.name}-heartbeat`, value);
+  }
+
+  counts() {
+    return this.client.call('queues', this.name).then(JSON.parse);
+  }
+
+  pause() {
+    return this.client.call('pause', this.name);
   }
-  scheduled(start, count, cb) {
-    this.client.call('jobs', 'scheduled', this.name, start || 0, count || 25, cb);
+
+  unpause() {
+    return this.client.call('unpause', this.name);
   }
 
-  // possible opts: jid, delay, priority, tags, retries, depends
-  put(klassName, data, opts, cb) {
-    this.client.call('put', this.workerName, this.name,
-        opts.jid || util.generateJid(),
-        klassName,
-        JSON.stringify(data),
-        opts.delay || 0,
-        'priority', opts.priority || 0,
-        'tags', JSON.stringify(opts.tags || []),
-        'retries', opts.retries || 5,
-        'depends', JSON.stringify(opts.depends || []),
-        cb);
+  /**
+   * Make a job
+   *
+   * The following attributes are required:
+   *  - klass
+   *
+   * The following attributes are optional:
+   *  - jid (defaults to a uuid)
+   *  - data (defaults to {}, must be JSON-serializable)
+   *  - priority (higher is more urgent)
+   *  - delay (in seconds)
+   *  - tags (array of strings)
+   *  - retries (how many times each job can be retried)
+   *  - depends (array of string jids on which this job depends)
+   */
+  put(options) {
+    // TODO(dan): should this yell if you provide an unsupported option?
```

Comment:
Regarding this comment, I would ask "how often has the job format changed, and do you want to allow independent evolution of the job format across language bindings?"